pgmpy/factors/Factor.py

```diff
@@ -770,9 +770,14 @@ def __eq__(self, other):
                 return True
 
     def __hash__(self):
-        return hash(str(self.variables) + str(self.cardinality.tolist()) +
-                    str(self.values.astype('float').tolist()))
-
+        value_index = 0
```

Comment:
@ankurankan I was also thinking for the same , but I believe this method happens to be faster .